src/app/pages/swap/hooks/use-bitflow-swappable-assets.tsx

```diff
@@ -0,0 +1,86 @@
+import { useCallback } from 'react';
+
+import { useQuery } from '@tanstack/react-query';
+import { Currency } from 'alex-sdk';
+import BigNumber from 'bignumber.js';
+import type { Token } from 'bitflow-sdk';
+
+import { createMarketData, createMarketPair } from '@leather.io/models';
+import {
+  type SwapAsset,
+  useAlexCurrencyPriceAsMarketData,
+  useAlexSdkLatestPricesQuery,
+  useStxAvailableUnlockedBalance,
+  useTransferableSip10Tokens,
+} from '@leather.io/query';
+import {
+  convertAmountToFractionalUnit,
+  createMoney,
+  getPrincipalFromContractId,
+  isDefined,
+  sortAssetsByName,
+} from '@leather.io/utils';
+
+import { createGetBitflowAvailableTokensQueryOptions } from '@app/query/bitflow-sdk/bitflow-available-tokens.query';
+
+export const BITFLOW_STX_CURRENCY: Currency = 'token-stx' as Currency;
+
+function useCreateSwapAsset(address: string) {
+  const { data: prices } = useAlexSdkLatestPricesQuery();
+  const priceAsMarketData = useAlexCurrencyPriceAsMarketData();
+  const availableUnlockedBalance = useStxAvailableUnlockedBalance(address);
+  const sip10Tokens = useTransferableSip10Tokens(address);
+
+  return useCallback(
+    (token?: Token): SwapAsset | undefined => {
+      if (!prices) return;
+      if (!token) return;
+      if (!token.tokenContract) return;
+
+      const principal = token.tokenContract;
+
+      const availableBalance = sip10Tokens.find(
+        sip10Token => getPrincipalFromContractId(sip10Token.info.contractId) === token.tokenContract
+      )?.balance.availableBalance;
+
+      const swapAsset = {
+        currency: token.tokenId as Currency,
+        fallback: token.symbol.slice(0, 2),
```

Comment:
not clear what this does? Helper function might explain better?

Reply:
I left a comment somewhere to revisit `SwapAsset` so that it before follows patterns we use with building our crypto assets in other parts of the app. Def a good idea to revisit once in the mono repo.

Reply:
I left a comment somewhere to revisit SwapAsset so that it before follows patterns we use with building our crypto assets in other parts of the app. Def a good idea to revisit once in the mono repo.